arm-operationalinsights/2015-03-20/swagger/OperationalInsights.json

```diff
@@ -0,0 +1,1110 @@
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "#/parameters/SubscriptionIdParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK response definition.",
+            "schema": {
+              "$ref": "#/definitions/SearchResultsResponse"
+            }
+          }
+        }
+      }
+    }
+  },
+  "definitions": {
+    "LinkTarget": {
+      "properties": {
+        "customerId": {
+          "type": "string",
+          "description": "The GUID that uniquely identifies the workspace. "
+        },
+        "accountName": {
+          "type": "string",
+          "description": "The display name of the workspace.",
+          "x-ms-client-name": "DisplayName"
+        },
+        "workspaceName": {
+          "type": "string",
+          "description": "The DNS valid workspace name."
+        },
+        "location": {
+          "type": "string",
+          "description": "The location of the workspace."
+        }
+      },
+      "description": "Metadata for a workspace that isn't linked to an Azure subscription."
+    },
+    "Tag": {
+      "properties": {
+        "Name": {
+          "type": "string",
+          "description": "The tag name."
+        },
+        "Value": {
+          "type": "string",
+          "description": "The tag value."
+        }
+      },
+      "required": [
+        "Name",
+        "Value"
+      ],
+      "description": "A tag of a saved search."
+    },
+    "SavedSearchProperties": {
+      "properties": {
+        "Category": {
+          "type": "string",
+          "description": "The category of the saved search. This helps the user to find a saved search faster. "
+        },
+        "DisplayName": {
+          "type": "string",
+          "description": "Saved search display name."
+        },
+        "Query": {
+          "type": "string",
+          "description": "The query expression for the saved search. Please see https://docs.microsoft.com/en-us/azure/log-analytics/log-analytics-search-reference for reference."
+        },
+        "Version": {
+          "type": "integer",
+          "format": "int64",
+          "minimum": 1,
+          "maximum": 1,
+          "description": "The version number of the query lanuage. Only verion 1 is allowed here."
+        },
+        "Tags": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/Tag"
+          },
+          "description": "The tags attached to the saved search."
+        }
+      },
+      "required":[
+        "Category",
+        "DisplayName",
+        "Query",
+        "Version"
+      ],
+      "description": "Value object for saved search results."
+    },
+    "SavedSearchCreateOrUpdateParameters": {
+      "properties": {
+        "etag": {
+          "type": "string",
+          "description": "The etag of the saved search."
+        },
+        "properties": {
```

Comment:
please mark properties required. This can have effects on flattening. For clients that do not flatten, properties will be an optional property in the payload. Which is clearly not the case.